addons/account_generator/models/generator.py

```diff
@@ -0,0 +1,212 @@
+
+    number_to_generate = fields.Integer(default=10)
+    number_of_lines = fields.Integer(default=10)
+    customer = fields.Boolean(default=True)
+    supplier = fields.Boolean(default=True)
+    company_type = fields.Selection([('person', 'Person'), ('company', 'Company')], default='person')
+    post = fields.Boolean()
+    end_date = fields.Date(default=fields.Date.today() + datetime.timedelta(days=30 * 3))
+    start_date = fields.Date(default=fields.Date.today() + datetime.timedelta(days=-365 * 3))
+    empty_field_probability = fields.Float(default=0.7)
+
+    company_id = fields.Many2one('res.company')
+    partner_ids = fields.Many2many('res.partner', relation="generator_partner_pool")
+    account_ids = fields.Many2many('account.account', relation="generator_account_pool")
+    journal_ids = fields.Many2many('account.journal', relation="generator_journal_pool")
+    partner_excluded_ids = fields.Many2many('res.partner', relation="generator_excluded_partner")
+    account_excluded_ids = fields.Many2many('account.account', relation="generator_excluded_account")
+    journal_excluded_ids = fields.Many2many('account.journal', relation="generator_excluded_journal")
+
+    ###### Wizard usability ####################################################
+
+    @api.onchange('empty_field_probability')
+    def _onchange_probability(self):
+        if not (0 <= self.empty_field_probability <= 1):
+            raise UserError(_('Probability must be between 0 and 1'))
+
+    ###### Generation functions ################################################
+
+    def generate_amls(self):
+        def generate_line(amount=None):
+            amount = amount or random.uniform(-10000, 10000)
+            return (0, 0, {'debit': amount > 0 and amount or 0, 'credit': amount < 0 and -amount or 0, 'account_id': random.choice(account_ids).id, 'partner_id': random.choice(partner_ids).id})
+
+        partner_ids, account_ids, journal_ids = self._get_records()
+        self._check_records(locals())
+        create_vals = []
+        for i in range(self.number_to_generate):
+            date = fake.date_time_between_dates(datetime_start=self.start_date, datetime_end=self.end_date)
+            create_vals.append({
+                'line_ids': [generate_line() for i in range(random.randint(1, self.number_of_lines - 1))],
+                'date': date,
+                'journal_id': random.choice(journal_ids).id,
+            })
+            amount = sum(line[2]['debit'] - line[2]['credit'] for line in create_vals[-1]['line_ids'])
+            create_vals[-1]['line_ids'].append(generate_line(-amount))
+        line_ids, view = self._create_records('account.move', create_vals)
+        if self.post:
+            line_ids.post()
+            _logger.info('Journal Entries posted')
+        return view
+
+    def generate_partners(self):
+        create_vals = []
+        for i in range(self.number_to_generate):
+            create_vals.append({
```

Comment:
would be nice to see all the val acquisitions split out to _prepare_* methods if this is for any kind of demo usefuleness